### PR TITLE
[MU4] Fix (MSVC) compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
reg. size_t vs. int, after #11194

### CODE REVIEW

File: src/framework/global/containers.h [hunk 1]
```diff
@@ -61,6 +63,7 @@ inline T value(const std::vector<T>& vec, size_t idx)
         return nullptr;
     }
 
+    UNREACHABLE;
     return T();
```
Comment (igorkorsukov) [inline src/framework/global/containers.h:67]: it can be reachable, I think better add 'else`

File: src/engraving/libmscore/scoretree.cpp [hunk 2]
```diff
@@ -874,7 +874,7 @@ EngravingObject* BSymbol::scanChild(int idx) const
 
 int BSymbol::scanChildCount() const
 {
-    return _leafs.size();
+    return static_cast<int>(_leafs.size());
```
Comment (igorkorsukov) [inline src/engraving/libmscore/scoretree.cpp:877]: I believe, better change method signature to `size_t BSymbol::scan ChildCount() const`

File: src/engraving/libmscore/part.cpp [hunk 3]
```diff
@@ -229,7 +229,7 @@ void Part::write(XmlWriter& xml) const
 
 int Part::nstaves() const
 {
-    return _staves.size();
+    return static_cast<int>(_staves.size());
```
Comment (igorkorsukov) [inline src/engraving/libmscore/part.cpp:232]: I believe, better change method signature to `size_t Part::nstaves() const`

File: src/engraving/libmscore/harmony.cpp [hunk 4]
```diff
@@ -2010,7 +2010,7 @@ void Harmony::addDegree(const HDegree& d)
 
 int Harmony::numberOfDegrees() const
 {
-    return _degreeList.size();
+    return static_cast<int>(_degreeList.size());
```
Comment (igorkorsukov) [inline src/engraving/libmscore/harmony.cpp:2013]: I believe, better change method signature to `size_t Harmony::numberOfDegrees() const`

File: src/framework/global/containers.h (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/framework/global/containers.h:67]: Hmm, MSVC apparently thinks it is not? And an else branch would equally (be reported as) unreachable?

No, apparently not, so I'll go for the `else`

File: src/engraving/libmscore/scoretree.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/scoretree.cpp:877]: I general I tend to agree, but that requires more changes in lots other places too

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/scoretree.cpp]: See b52db31

File: src/engraving/libmscore/part.cpp (hunk 3 above)
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/part.cpp:232]: Same comment as above, many more changes needed in lots other places

File: src/engraving/libmscore/harmony.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/harmony.cpp:2013]: This one is an easy change, will do.

File: src/engraving/libmscore/scoretree.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/scoretree.cpp]: You wanna go that way?

File: src/engraving/libmscore/part.cpp (hunk 3 above)
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/part.cpp:232]: This one seems even worse, requiring more casting elsewhere (or even much bigger changes)

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/part.cpp:232]: Not worth the effoert IMHO

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/part.cpp:232]: not leading to cleaner code at a reasonable price

File: src/engraving/libmscore/scoretree.cpp (hunk 2 above)
Comment (igorkorsukov) [reply src/engraving/libmscore/scoretree.cpp]: I believe it's worth it

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/scoretree.cpp]: This one indeed might. Also it is done ;-)
Not so the other one, below.

File: src/engraving/libmscore/part.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/engraving/libmscore/part.cpp:232]: I think it's better to do this later when most of the code will be changed to std

**Comment by Jojo-Schmitz** [comment]
Hmm, that merge was a bit early, I was just about to check whether any of the changes cause issues/warnings with MinGW, and sure enough this is the case..
Oh well, next PR coming...

**Comment by Jojo-Schmitz** [comment]
Better revert that last commit, it just causes tons of warnings with MinGW (and presumable other gcc based compilers).
Fixing those would be another huge change, esp. when trying to avoid introducing other MSVC warnings (and on fixing those causeing other MinGW warnings)

**Comment by igorkorsukov** [comment]
> Hmm, that merge was a bit early, I was just about to check whether any of the changes cause issues/warnings with MinGW, and sure enough this is the case..
> Oh well, next PR coming...

Oh, sorry, I merged quickly to reduce conflicts (

**Comment by Jojo-Schmitz** [comment]
My fault (too) I guess, I should have marked it DRAFT...
I also would have squashed the 3rd commit into the 1st. But now I'm glad I did not, makes reverting that 3rd commit far easier.
I believe we're better of witout it, for now, and possibly revisit all these size_t issues later,  _when most of the code will be changed to std_

**Comment by igorkorsukov** [comment]
I looked, commit revert not need, I try fix most warning in a next PR

**Comment by Jojo-Schmitz** [comment]
There are also still tons of warnings with MSVC, not sure why (or rather why I hadn't seen them earlier).
And even if I do revert that commit! I'm severely confused now.

**Pull request merged**